HeartBeat.php

```diff
@@ -75,31 +70,44 @@ public function newDirectRequest()
      */
     public function getPluginSettings($getValues=true)
     {
-        /* We fix the default before read the parent: the arent set the curretnt to actual default */
-        $this->settings['interval']['default']=self::getDefaultInterval();
-        $this->settings['interval']['help']="Your actual session.gc_maxlifetime is ".App()->session->getTimeout()."s";
+        /* We fix the default before read the parent: the parent set the current to actual default */
+        $this->settings['interval']['default'] = self::getDefaultInterval();
+        $this->settings['useSessionLifeTime']['help'] = sprintf($this->_translate("Your actual session.gc_maxlifetime is %s seconds"),App()->session->getTimeout());
+        $this->settings['interval']['label'] = $this->_translate("Interval of the heartbeat (in seconds, minimum: 5s)");
         $aPluginSettings = parent::getPluginSettings($getValues);
 
         $useSessionLifeTime = (boolean) $this->get('useSessionLifeTime', null, null, false);
-
-        // If it is a post request, it could be an autosubmit so read posted
-        // value over the saved value
-        if (App()->request->isPostRequest)
-        {
+        if (App()->request->isPostRequest) {
             $useSessionLifeTime = App()->request->getPost('useSessionLifeTime', $useSessionLifeTime);
+            if(empty($useSessionLifeTime) && intval(App()->request->getPost('interval')) < 5) {
+                /* Saved in saveSettings, but call is done before */
+                $useSessionLifeTime = 1;
+            }
         }
-
-        if($useSessionLifeTime == true){
-            unset($aPluginSettings['interval']);
+        if($useSessionLifeTime) {
+            $this->settings['interval']['help'] = sprintf($this->_translate("This value is not used, current value used : %s seconds."),$this->getDefaultInterval());
+            $this->settings['interval']['htmlOptions']['placeholder'] = $this->getDefaultInterval();
         }
         return $aPluginSettings;
     }
 
     /**
-     * Default value for interval
+     * Fix when saving
+     * @param type $settings
      */
-    private function getDefaultInterval()
+    public function saveSettings($settings)
     {
+        if(empty($settings['useSessionLifeTime']) && intval($settings['interval']) < 5) {
+            $this->set('useSessionLifeTime', 1);
+            return;
```

Comment:
If I understand this correctly, every time the user doesn't want to use sessionLifeTime, but saves an interval smaller than 5seconds, it will automatically check the useSessionLifeTime option?

Why is this behavior necessary? Can't we provide a minimum value for the input so that the user can't save the form when the value is lower than 5 seconds ?

Reply:
Yes, we can put a min, but a min can be disable via source edit.

HTML control is done via https://github.com/frederikprijck/limesurvey-plugin-heartbeat/blob/1dd3e42d30b46c349f82c9174626bd118f430425/HeartBeat.php#L31
PHP (server) control must always be done

But : we force 5 here too : https://github.com/frederikprijck/limesurvey-plugin-heartbeat/blob/a61c39a80a59b90802ddde9a93c2e11acf34f39f/HeartBeat.php#L53

We can change and set to 5 if you want :)

Reply:
> Yes, we can put a min, but a min can be disable via source edit.

Ofcourse, it's frontend. It's not security, But it's better for the user experience.

So does this mean we do not need to set the useSessionLifeTime to 1 incase the interval is smaller than 5? Instead, we'll use validation force an interval gt 5 ?

Reply:
Looking at code, i think we can't put a value lesser than 5 (min is here).
But we can leave empty for the both : then i think it check automatic if both are empty.
Let le check, maybe

1. Both are empty : set it to automatic
2. Not empty value but lesser than 5 => put 5

Reply:
Sounds good!